### ISSUE TITLE
Linking failure (due to -Werror) with LTO

### ISSUE THREAD

**Issue by ArchangeGabriel** [intel/intel-graphics-compiler#237]
When compiling with LTO, the build fails like this:
```
/build/intel-graphics-compiler/src/intel-graphics-compiler-igc-1.0.10778/visa/Timer.cpp:80: error: type ‘struct Timer’ violates the C++ One Definition Rule [-Werror=odr]
   80 | struct Timer {
      | 
/build/intel-graphics-compiler/src/intel-graphics-compiler-igc-1.0.10778/visa/iga/IGALibrary/Timer/Timer.cpp:76: note: a different type is defined in another translation unit
   76 | struct Timer {
      | 
/build/intel-graphics-compiler/src/intel-graphics-compiler-igc-1.0.10778/visa/Timer.cpp:86: note: the first difference of corresponding definitions is field ‘hits’
   86 |     unsigned int hits;
      | 
/build/intel-graphics-compiler/src/intel-graphics-compiler-igc-1.0.10778/visa/iga/IGALibrary/Timer/Timer.cpp:76: note: a type with different number of fields is defined in another translation unit
   76 | struct Timer {
      | 
```

**Comment by ArchangeGabriel** [comment]
Some new ones:
```
/build/intel-graphics-compiler/src/intel-graphics-compiler-igc-1.0.11485/IGC/VectorCompiler/include/vc/Utils/GenX/Region.h:256:7: error: type ‘struct Region’ violates the C++ One Definition Rule [-Werror=odr]
  256 | class Region : public CMRegion {
      |       ^
/usr/include/llvm/Analysis/RegionInfo.h:889: note: a type with different bases is defined in another translation unit
  889 | class Region : public RegionBase<RegionTraits<Function>> {
      | 
/build/intel-graphics-compiler/src/intel-graphics-compiler-igc-1.0.11485/IGC/VectorCompiler/lib/GenXCodeGen/GenXRegionUtils.h:79:8: error: ‘makeRegionWithOffset’ violates the C++ One Definition Rule [-Werror=odr]
   79 | Region makeRegionWithOffset(const Instruction *Inst,
      |        ^
/build/intel-graphics-compiler/src/intel-graphics-compiler-igc-1.0.11485/IGC/VectorCompiler/lib/GenXCodeGen/GenXRegionUtils.cpp:116: note: ‘makeRegionWithOffset’ was previously declared here
  116 | Region genx::makeRegionWithOffset(const Instruction *Inst,
      | 
/build/intel-graphics-compiler/src/intel-graphics-compiler-igc-1.0.11485/IGC/VectorCompiler/lib/GenXCodeGen/GenXRegionUtils.h:82:8: error: ‘makeRegionFromBaleInfo’ violates the C++ One Definition Rule [-Werror=odr]
   82 | Region makeRegionFromBaleInfo(const Instruction *Inst, const BaleInfo &BI,
      |        ^
/build/intel-graphics-compiler/src/intel-graphics-compiler-igc-1.0.11485/IGC/VectorCompiler/lib/GenXCodeGen/GenXRegionUtils.cpp:144: note: ‘makeRegionFromBaleInfo’ was previously declared here
  144 | Region genx::makeRegionFromBaleInfo(const Instruction *Inst, const BaleInfo &BI,
      | 
```
I’ll disable the erroring in order to try to get all of them.

**Comment by ArchangeGabriel** [comment]
Just another one:
```
/build/intel-graphics-compiler/src/intel-graphics-compiler-igc-1.0.11485/IGC/GenISAIntrinsics/GenIntrinsics.cpp:92: warning: type ‘IIT_Info’ violates the C++ One Definition Rule [-Wodr]
   92 | enum IIT_Info {
      | 
/build/intel-graphics-compiler/src/intel-graphics-compiler-igc-1.0.11485/IGC/VectorCompiler/lib/InternalIntrinsics/InternalIntrinsics.cpp:58: note: an enum with different value name is defined in another translation unit
   58 | enum IIT_Info {
      | 
/build/intel-graphics-compiler/src/intel-graphics-compiler-igc-1.0.11485/IGC/GenISAIntrinsics/GenIntrinsics.cpp:129: note: name ‘IIT_VEC_OF_PTRS_TO_ELT’ differs from name ‘IIT_PTR_TO_ELT’ defined in another translation unit
  129 |   IIT_VEC_OF_PTRS_TO_ELT = 33,
      | 
/build/intel-graphics-compiler/src/intel-graphics-compiler-igc-1.0.11485/IGC/VectorCompiler/lib/InternalIntrinsics/InternalIntrinsics.cpp:95: note: mismatching definition
   95 |   IIT_PTR_TO_ELT = 33,
      | 
```

**Comment by ArchangeGabriel** [comment]
Only the last batch was fixed by this commit, the ones from https://github.com/intel/intel-graphics-compiler/issues/237#issuecomment-1168432408 still remain.

**Comment by Pitchiu** [comment]
> Only the last batch was fixed by this commit, the ones from [#237 (comment)](https://github.com/intel/intel-graphics-compiler/issues/237#issuecomment-1168432408) still remain.

This was fixed by https://github.com/intel-innersource/drivers.gpu.unified/pull/78626
